File: solutions-py/src/main/array/LC_0001_two_sum.py
from typing import List

class TwoSum:
  def twoSum(self, nums: List[int], target: int) -> List[int]:
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    _dict = {}
    for i, num in enumerate(nums):
      if target-num in _dict.keys():
        return [_dict[target-num], i]
      else:
        _dict[num] = i

File: solutions-go/src/main/array/LC_0001_two_sum.go
package array;

func TwoSum(nums []int, target int) []int {
    numMap := make(map[int]int)
    for i, num := range nums {
        if j, ok := numMap[target-num]; ok {
            return []int{j, i}
        }
        numMap[num] = i
    }
    return nil
}
